### ISSUE TITLE
undefined method `version' for nil:NilClass (NoMethodError)

### ISSUE THREAD

**Issue by jcope2013** [MaximeD/gem_updater#12]
```
gem install gem_updater
gem_update
Updating gems...

... (Updating gems)

Bundle updated!
/Users/jcope/.rvm/gems/ruby-2.1.2@inboxhealth/gems/gem_updater-0.1.0/lib/gem_updater/gem_file.rb:28:in `block in compute_changes': undefined method `version' for nil:NilClass (NoMethodError)
	from /Users/jcope/.rvm/gems/ruby-2.1.2@inboxhealth/gems/gem_updater-0.1.0/lib/gem_updater/gem_file.rb:27:in `compute_changes'
	from /Users/jcope/.rvm/gems/ruby-2.1.2@inboxhealth/gems/gem_updater-0.1.0/lib/gem_updater/gem_file.rb:20:in `update!'
	from /Users/jcope/.rvm/gems/ruby-2.1.2@inboxhealth/gems/gem_updater-0.1.0/lib/gem_updater.rb:17:in `update!'
	from /Users/jcope/.rvm/gems/ruby-2.1.2@inboxhealth/gems/gem_updater-0.1.0/bin/gem_update:9:in `<top (required)>'
	from /Users/jcope/.rvm/gems/ruby-2.1.2@inboxhealth/bin/gem_update:23:in `load'
	from /Users/jcope/.rvm/gems/ruby-2.1.2@inboxhealth/bin/gem_update:23:in `<main>'
	from /Users/jcope/.rvm/gems/ruby-2.1.2@inboxhealth/bin/ruby_executable_hooks:15:in `eval'
	from /Users/jcope/.rvm/gems/ruby-2.1.2@inboxhealth/bin/ruby_executable_hooks:15:in `<main>
```

**Comment by MaximeD** [comment]
Hello, can you provide your original `Gemfile.lock` (in it's state before update) so that I can try to reproduce the bug ? Thanks

**Comment by jrgifford** [comment]
I think it's because of pulling gems from git.

**Comment by mrfoto** [comment]
Yup I get the same. Fetching gems directly from GitHub seems to be the issue.

**Comment by MaximeD** [comment]
I don't know, I have a line like this in my `Gemfile`:
```
 gem 'foo', github: 'example/foo'
```

Downgrading it to a previous revision in my `Gemfile.lock` and launching `gem_update` does not raise any errors:

```
GIT
  remote: git://github.com/example/foo.git
  revision: <old_commit>
```

Though it does not try to get a changelog for this gem.

Can you provide a minimal failing example?
Thank you

**Comment by mrfoto** [comment]
It's hard to provide minimal example because it depends on the `.lock` file as well.

I debugged with:
```
unless @new_spec_set[ gem_specs.name ].first.respond_to? :version
  p "#{gem_specs.name}: #{@new_spec_set[ gem_specs.name ]}"
  next
end
```

and got `"rspec-instafail: []"` which is not present in the `Gemfile` but it is in `Gemfile.lock` as a dependency of [`fuubar 1.2.1`](https://github.com/thekompanee/fuubar/tree/9e0e0311ae56656ff8b13e184769f07df3c6730b).

It looks like git is not to blame but getting a version of something that is not in Gemifile.lock anymore, but was before.

**Comment by MaximeD** [comment]
Indeed, this happens when a `gem` is removed from `Gemfile.lock`, and `fuubar` removed dependency to `rspec-instafail` (which happened [here](https://github.com/thekompanee/fuubar/commit/c7d03fde7d9275f8f38523536005ae0465c7af84#diff-9d91df34bd8b406a7191b78d17daf2d7L22)). Thanks for tracking it, I'll have a look.

**Comment by mrfoto** [comment]
:+1:

**Comment by MaximeD** [comment]
Hello, this has been fix in version 0.2.0

**Comment by mrfoto** [comment]
Thanks! :+1:
